torch/lib/c10d/CMakeLists.txt

```diff
@@ -23,18 +34,36 @@ else()
   MESSAGE(STATUS "Not able to find MPI, will compile c10d without MPI support")
 endif()
 
+find_package(CUDA REQUIRED)
+if(NOT CUDA_FOUND)
+  message(FATAL_ERROR "CUDA not found")
```

Comment:
Do we want to fail the build if CUDA cannot be found? I am thinking of making it optional

Reply:
Right now I think it is acceptable, so we can get a first working version out the door.